src/tests/unit/ingredients-dao.test.js

```diff
@@ -0,0 +1,145 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import config from 'config';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+import { getIngredientsData } from 'tests/unit/test-data';
+import { openapi } from 'utils/load-openapi';
+import { GetFilterProcessor } from 'utils/process-get-filters';
+
+const ingredientsGetParameters = openapi.paths['/ingredients'].get.parameters;
+
+sinon.restore();
+
+sinon.replace(config, 'get', () => ({ oracledb: {} }));
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+describe('test ingredients dao', () => {
+  const {
+    getIngredientsQuery,
+    emptyGetIngredientsQuery,
+    testFilters,
+    ingredientAliases,
+    testBindParams,
+    testSerializerReturn,
+    testConditionals,
+    testConnectionReturn,
+    testConnectionReturnRows,
+  } = getIngredientsData;
+  let serializeIngredientsStub;
+  let getConnectionStub;
+  let connectionSpy;
+  let ingredientsDao;
+  let getFilterProcessorStub;
+  let getFilterProcessorContructorStub;
+  beforeEach(() => {
+    connectionSpy = sinon.stub().returns(testConnectionReturn);
+    serializeIngredientsStub = sinon.stub().returns(testSerializerReturn);
+    getConnectionStub = sinon.stub().resolves({
+      execute: connectionSpy,
+      close: () => {},
+    });
+    getFilterProcessorContructorStub = sinon.stub();
+    getFilterProcessorStub = sinon.stub(GetFilterProcessor.prototype, 'processGetFilters').returns({
+      bindParams: testBindParams,
+      conditionals: testConditionals,
+    });
+
+    Object.setPrototypeOf(GetFilterProcessor, getFilterProcessorContructorStub);
+
+    ingredientsDao = proxyquire('../../api/v1/db/oracledb/ingredients-dao', {
+      './connection': {
+        getConnection: getConnectionStub,
+      },
+      '../../serializers/ingredients-serializer': {
+        serializeIngredients: serializeIngredientsStub,
+      },
+    });
+  });
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe('when the constructor for GetFilterProcessor is called', () => {
+    beforeEach(() => {
+      ingredientsDao = proxyquire('../../api/v1/db/oracledb/ingredients-dao', {
+        './connection': {
+          getConnection: getConnectionStub,
+        },
+        '../../serializers/ingredients-serializer': {
+          serializeIngredients: serializeIngredientsStub,
+        },
+        '../../../../utils/process-get-filters': {
+          GetFilterProcessor: getFilterProcessorContructorStub,
+        },
```

Comment:
This is stubbed with proxyquire in only this test because stubbing objects in sinon is, as it turns out, a pain. Because the constructor has no side effects outside of the class and the only public method of the class is stubbed, there's no functional difference between this approach and one that more thoroughly stubs out the class in every function, besides that this approach is more concise.